src/fetch.ts

```diff
@@ -168,18 +168,27 @@ export function createFetch(globalOptions: CreateFetchOptions = {}): $Fetch {
 
     let abortTimeout: NodeJS.Timeout | undefined;
 
-    // TODO: Can we merge signals?
-    if (!context.options.signal && context.options.timeout) {
-      const controller = new AbortController();
-      abortTimeout = setTimeout(() => {
-        const error = new Error(
-          "[TimeoutError]: The operation was aborted due to timeout"
+    if (context.options.timeout) {
+      if (context.options.signal) {
+        context.options.signal = AbortSignal.any(
+          [
+            AbortSignal.timeout(context.options.timeout),
+            context.options.signal,
+            // eslint-disable-next-line unicorn/prefer-native-coercion-functions
+          ].filter((s): s is NonNullable<typeof s> => Boolean(s))
```

Comment:
We already checked in L171 and L172 that both are non nullable i think filter is extra here.